### PR TITLE
Add Security/XML sections to Technical Guidelines

### PR DESCRIPTION
**Pull request by keharper**
## This PR is a:

- [ ] New topic
- [x] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged: t
* The Security section of the Technical Guides will be added to the 2.0 and 2.1 branches. (The section is currently in the 2.2 and 2.3 branches.)
* The XML configuration section will be added to the 2.1, 2.2. and 2.3 branches. This item originates from [PR-2673](https://github.com/magento/devdocs/pull/2673/)

### CODE REVIEW

File: guides/v2.1/coding-standards/technical-guidelines.md [hunk 1]
```diff
@@ -697,6 +697,121 @@ class SampleEventObserverThatModifiesInputs
...
 14.2. Events used SHOULD be observed as specifically as possible. A `global` subscription to an event SHOULD NOT be used when the area impacted is just `frontend`.
 
+## 15. XML configuration
+
+15.1 Attributes MUST be used for the configuration values that will not be extended in the future by adding child nodes or new attributes.
+
+15.2 Nodes of the same type that repeat multiple times MUST be wrapped in a grouping node.
+
+15.3 Unnecessarily grouping nodes MUST be avoided.
+
```
Comment (keharper) [inline guides/v2.1/coding-standards/technical-guidelines.md]: The left and right columns in the table are identical, except for the comment. Is this intentional?

File: guides/v2.1/coding-standards/technical-guidelines.md [hunk 2]
```diff
@@ -697,6 +697,121 @@ class SampleEventObserverThatModifiesInputs
...
+    </tr>
+    <tr>
+        <td>
+{% highlight xml %}
+<!-- This example assumes that the following configuration is in its own file and `item` doesn't repeat multiple times in the file -->
+<config>
+    <items>
+        <item />
+    </items>
+<config>
```
Comment (keharper) [inline guides/v2.1/coding-standards/technical-guidelines.md]: Should this be `</config>`?

**Comment by keharper** [comment]
@buskamuza @melnikovi a review draft is at

http://cicd1.corp.magento.com/~mmansoor/devdocs/gh-pages/branches/kh_tech_guidelines/_site/guides/v2.1/coding-standards/technical-guidelines.html

**Comment by magento-cicd2** [comment]
Can one of the admins verify this patch?

**Comment by jeff-matthews** [comment]
@keharper, please finish processing this PR and merge it.

File: guides/v2.1/coding-standards/technical-guidelines.md (hunk 2 above)
Comment (osrecio) [reply guides/v2.1/coding-standards/technical-guidelines.md]: Yes, I can confirm it

**Comment by jeff-matthews** [comment]
@keharper, is this waiting for an internal architect review?

**Comment by hguthrie** [comment]
There has been no updates on this PR for some time. Closing this PR.

**Pull request closed without merging**